File: 2019/Go/day1.go
package main

import (
	"fmt"
	"bufio"
	"os"
	"strconv"
)

func gas(x int) int {
	return x / 3 - 2
}
func part_one() {
	sum_ := 0
	file, _ := os.Open("../python/day1.input")
	scanner := bufio.NewScanner(file)
    for scanner.Scan() {
        i, _ := strconv.Atoi(scanner.Text())
        sum_ += gas(i) 

    }
    fmt.Println(sum_)
}

func part_two() {
	sum_ := 0
	file, _ := os.Open("../python/day1.input")
	scanner := bufio.NewScanner(file)
    for scanner.Scan() {
        i, _ := strconv.Atoi(scanner.Text())
        temp := gas(i) 
        for temp > 0 {
        	sum_ += temp
        	temp = gas(temp)
        }
    }
    fmt.Println(sum_)
}

func main() {
	part_two()
}

File: 2019/Python/day1.py
from math import floor

def part_one():

	# print(sum([*map(lambda a : a//3 - 2,[int(i) for i in open('day1.input')])]))


	a = lambda a: a//3-2

	with open('day1.input') as file:
		b = [a(int(i)) for i in file]
		print(sum(b))


def part_two():

	
	a = lambda a: floor(a/3)-2
	all_sum = 0

	with open('day1.input') as file:
		for i in file:
			t = a(int(i))
			sum_ = t
			while a(int(t)) > 0:
				t = a(int(t))
				sum_ += t
			all_sum += sum_
		print(all_sum)

if __name__ == '__main__':
	part_one()
	part_two()